### ISSUE TITLE
Lights turn back on when turned off manually

### ISSUE THREAD

**Issue by phoniclynx** [basnijholt/adaptive-lighting#343]
#### Description:
Whenever I turn off a light a few seconds later that light or one of the lights in a group that was turned off decided it wants to turn back on.Turning off the adaptive light this issue doesn't appear. Lights are a LIFX bulb This sometimes happens with an automation of a PIR turning off a light when they detect no motion. They will just turn back on at random.

**Comment by utsnik** [comment]
Having the same issue :/

**Comment by dkitsov** [comment]
Same with Philips Hue integration lights.

**Comment by basnijholt** [comment]
Did this always happen or only since recent updates?

**Comment by dkitsov** [comment]
I would say within the last 5–7 days.

**Comment by basnijholt** [comment]
The latest update was 25 days ago.

Are you saying that this started happening since you updated Adaptive Lighting from an older version to the current version? Or has this always happened?

One thing is that if you are using Zigbee groups, this problem is more likely to occur.

**Comment by phoniclynx** [comment]
It's quite possible it was 25 days ago. due to my job, I don't really have a good perception of time (long story) but it is interesting that you say Zigbee device groups could affect this as there is one light that is a Zigbee light but not in a group as such. Just a stand-alone light. I'll try taking it out of the adaptive lighting and see if it makes a difference tonight

**Comment by phoniclynx** [comment]
I took it out of the group and restarted it and it didn't work: 
![image](https://user-images.githubusercontent.com/3883152/192665932-27fdb48f-b196-45b7-8d47-d1cea8ee1c44.png)

**Comment by daFreeMan** [comment]
I've been having the exact same issue (Sengled bulbs). It's been happening to me for a _very_ long time, so it's nothing new with the latest release of AL.

I've uninstalled AL and the bulbs go off when requested and, more importantly, _stay_ off.

**Comment by dkitsov** [comment]
> 

I reverted to 1.0.19 a few days ago and did not observe the behavior since

**Comment by basnijholt** [comment]
@dkitsov, that is great to know!

That means that the problem must have been introduced in some of this code: https://github.com/basnijholt/adaptive-lighting/compare/1.0.19...1.1.0

Could someone else verify that using 1.0.19 solves the issue described here?

**Comment by daFreeMan** [comment]
Just saw this. I've installed 1.0.19 and will do some testing, but I'm not holding out hope. I was previously on 1.0.15 (I think) and it wasn't working for me. Maybe it was fixed in .0.19 and broken again by .1.0

**Comment by basnijholt** [comment]
The "real" problem is not with Adaptive Lighting I think. The lights are probably reporting to Home Assistant that they are briefly on after they are turned off.

**Comment by daFreeMan** [comment]
Three different brands of bulbs for three different users. 
My bulbs work fine when they're _not_ assigned in the AL integration (either as a group in the fixture or individually). 
It _is_ working in 1.0.19 (I turn them off and they stay off).

**Comment by BadCo-NZ** [comment]
Is this not a duplicate of #246 ?

A few of us are stuck on 1.0.14 due to this bug.

**Comment by ds-sebastian** [comment]
Ah I am also experiencing this, where lights are randomly turning on. I will try a different version to see if the issue stops

**Comment by BadCo-NZ** [comment]
1.0.19 seems to he working without the issue

**Comment by StSaens** [comment]
As a work around I'm turning AL off in the script that turns lights off. Going to see if this prevents the error from occuring. Apparantly my living's light were all on last night while we were asleep.

**Comment by phoniclynx** [comment]
I've only noticed this in the past month or so

**Comment by HStep20** [comment]
As others have mentioned, this has been an ongoing issue in my experience as well. Im pretty sure its related to #246, but all of my automations regarding lights have to turn off Adaptive Lighting because of this. It was the only/best solution Id found at the time of noticing the behavior and trying to get it fixed. It'd be nice to have it fixed, but I may try rolling back to 1.0.14 if it doesn't have the issue as badco-nz implied. 

All of my lights are Philips Hue going through the Hue Bridge

**Comment by daFreeMan** [comment]
I updated to 1.2.1 a couple of days ago and have not had issues with lights turning back on when I manually turn them off.

I don't think there was any intentional effort made, but maybe, _just maybe_ this has been resolved. I would suggest that others update and test out this latest release. Maybe I've gotten lucky, maybe it's actually fixed.

**Comment by basnijholt** [comment]
Interestingly it started happening for me now. But only after updating the firmware of my Shelly Dimmers.

I investigated and what happens is that the dimmer reports to HA that it is on (while it is not). Adaptive Lighting acts by changing the lights to the right color because it has just been turned on.

**Comment by chrisbanes** [comment]
This seems to have gotten worse for over the past 2 weeks or so (I assume due to upgrading 1.2.1). It initially started happening for me (and driving my wife crazy) when I migrated from deCONZ to ZHA.

**Comment by daFreeMan** [comment]
Ha! Serves ya' right!!!! LOL 

No, just kidding... I have a couple of Shelly 1, but no dimmers. This only happens to me on my Zigbee bulbs. Again, though, I was having no issues with it on 1.0.19, updated to 1.2.1 and still am not having issues.

Having other issues with light temp not matching between two sets of bulbs, both controlled by AL, but that's a different issue that I'll have to document thoroughly...

**Comment by GollyJer** [comment]
What can we do, as users, to help troubleshoot this? 
I've been following along for a while, and there seems to be no solution in sight. Are there logs or something we can provide?

**Comment by basnijholt** [comment]
Yes the debugging logs would be super helpful.

**Comment by basnijholt** [comment]
That is truly the only way that I can debug this. I would need to see what the locks say exactly when the lights turned on. Please post the logs from when it happened and just before it.

See https://github.com/basnijholt/adaptive-lighting/blob/master/README.md#having-problems

**Comment by basnijholt** [comment]
This is actually a duplicate of #10. I will post my findings there.
